### ISSUE TITLE
`tests/poisson_2d` example diverging when `useStrongDirichletConstraints==True`

### ISSUE THREAD

**Issue by mfarthin** [erdc/proteus#394]
I am probably not setting all of the correct flags, but when I derive a local dummy flux and set the class variable `useStrongDirichletConstraints==True`, the newton solve diverges for the linear poisson 2d example in `tests/poisson_2d` . This is with all Dirichlet boundaries or partial Dirichlet boundaries. 

One thing that may or may not be an possible issue is that the 
`self.dirichletConditions[cj].DOFBoundaryConditionsDict` has no keys (dofs), which I believe follows from the fact that `useWeakDirichletConditions==True` in the `MultilevelTransport` initialization by default when we use a numerical flux. 

When I check the residual calculation with forceStrongConditions==True the loop at the start of the `getResidual` routine in `OneLevelTransport`

```
    def getResidual(self,u,r):
        """
        Calculate the element residuals and add in to the global residual
        """
        #cek debug
        #u.tofile("u"+`self.nonlinear_function_evaluations`,sep="\n")
        r.fill(0.0)
        for cj in range(self.nc):
            for dofN,g in self.dirichletConditions[cj].DOFBoundaryConditionsDict.iteritems():
                self.u[cj].dof[dofN] = g(self.dirichletConditions[cj].DOFBoundaryPointDict[dofN],self.timeIntegration.t)
        if self.forceStrongConditions:
            for cj in range(len(self.dirichletConditionsForceDOF)):
                for dofN,g in self.dirichletConditionsForceDOF[cj].DOFBoundaryConditionsDict.iteritems():
                    u[self.offset[cj]+self.stride[cj]*dofN] = self.u[cj].dof[dofN]#load the BC value directly into the global array

```

The `self.u[cj].dof` isn't set to the Dirichlet value because the iteritems is empty and that follows for the `u` array. The `self.dirichletConditionsForceDOF` dictionary is used later on to set the residual and jacobian values. It has the complete list of Dirichlet degrees of freedom.

**Comment by cekees** [comment]
You're, right,  that last line should call the `g` function directly to set the `u` array, like this:

```
u[self.offset[cj]+self.stride[cj]*dofN] = g(self.dirichletConditionsForceDOF[cj].DOFBoundaryPointDict[dofN],self.timeIntegration.t)
```
 I think I must have done that at the last minute working on cleaning up and optimizing, thinking I was eliminating a second call to `g`, but it only gets called once anyway because the first  list of  BC's is empty.  

I suspect if you set the initial conditions consistent with the boundary conditions it hides this bug.  Could you add this test directory to https://github.com/erdc-cm/proteus/blob/master/.travis.yml#L35 and maybe add the case with `forceStrongConditions=True` to the `tests/poisson_2d`? These tests run quickly so there's no reason why they shouldn't be part of the tests travis runs.

**Comment by adimako** [comment]
Addressed in #396
